contracts/dao/FrabricDAO.sol

```diff
@@ -84,17 +94,32 @@ abstract contract FrabricDAO is EIP712Upgradeable, DAO, IFrabricDAO {
     return (pType >> 8) == 1;
   }
 
+  /**
+   * @notice Create a new proposal for a paper (statement agreed upon by the DAO)
+   * @param supermajority True if supermajority required for proposal to pass (binding propsal)
+   * @param info Paper statement to be proposed
+   * @return uint256 Id of created proposal
+   */
   function proposePaper(bool supermajority, bytes32 info) external override returns (uint256) {
     // No dedicated event as the DAO emits type and info
     return _createProposal(uint16(CommonProposalType.Paper) | commonProposalBit, supermajority, info);
   }
 
-  // Allows upgrading itself or any contract owned by itself
-  // Specifying any irrelevant beacon will work yet won't have any impact
-  // Specifying an arbitrary contract would also work if it has functions/
-  // a fallback function which don't error when called
-  // Between human review, function definition requirements, and the lack of privileges bestowed,
-  // this is considered to be appropriately managed
+  /**
+   * @notice Propose a contract upgrade, allows upgrading itself or any contract owned by itself
+   * @param beacon Address of beacon contract facilitating upgrades
+   * @param instance Address of contract instance to be upgraded
+   * @param version Version number
+   * @param code Address of contract containing new code
+   * @param data Data to be passed to new contract [?]
```

Comment:
Several parts of this contract were challenging to get right, someone more familiar should definitely check my @params are accurate here.